core/src/test/scala/io/projectglow/vcf/VCFFileWriterSuite.scala

```diff
@@ -400,6 +400,31 @@ abstract class VCFFileWriterSuite(val sourceName: String)
     assert(sampleIds.length == 1)
     assert(sampleIds.head == Seq("sample_1", "sample_2", "sample_3"))
   }
+
+  test("validate schema before write") {
```

Comment:
This would benefit from more thorough testing - at minimum, should iterate through each of the expected required fields. Ideally, we should have a sanity check to make sure that our expected required fields match the actual required fields.

Reply:
Done